ui.apps/src/main/javascript/minicart/src/context/UserContext.js

```diff
@@ -57,16 +64,45 @@ const UserContextProvider = props => {
             setUserState({ ...userState, signInError });
         }
     });
-    const [
-        getCustomerDetails,
-        { data: customerData, error: customerDetailsError, loading: customerDetailsLoading }
-    ] = useLazyQuery(QUERY_CUSTOMER_DETAILS);
 
     const [
         revokeCustomerToken,
         { data: revokeTokenData, error: revokeTokenError, loading: revokeTokenLoading }
     ] = useMutation(MUTATION_REVOKE_TOKEN);
 
+    const [
+        createCustomer,
+        { data: createCustomerData, error: createCustomerError, loading: createCustomerLoading }
+    ] = useMutation(MUTATION_CREATE_CUSTOMER, {
+        onError: error => {
+            let createAccountError = parseError(error);
+            setUserState({ ...userState, createAccountError });
+        }
+    });
+
+    /*
+     * We use the apollo client directly
+     */
+    const getCustomerDetails = opts => {
```

Comment:
Can you elaborate why you have to use the Apollo client directly here and can't use the hook you used before?

Reply:
I already added a comment (actually, I forgot to write the whole comment above 🙈) to explain this. I stumbled into an issue with the following flow: 
- create an account
- sign out
- create another account - the "My Account" panel showed the data of the first account. 

The reason was that the data returned by the first `getCustomerDetails` call was still available so that the code just returned that. This whole "hook-a-palooza" needs to be refactored a bit, maybe even using the client directly or just refactor the components to issue their own query.

Reply:
Got it, thanks :) Looks good for now, let's improve it later.